tests/node_tests/backbeat/tests/retry/pendingMetrics.js

```diff
@@ -0,0 +1,162 @@
+const assert = require('assert');
+const crypto = require('crypto');
+const { doWhilst, parallel, waterfall, series } = require('async');
+
+const { scalityS3Client, awsS3Client } = require('../../../s3SDK');
+const sharedBlobSvc = require('../../azureSDK');
+const ReplicationUtility = require('../../ReplicationUtility');
+const { makeGETRequest, getResponseBody, makePOSTRequest } =
+    require('../../../utils/request');
+
+const scalityUtils = new ReplicationUtility(scalityS3Client, sharedBlobSvc);
+const awsUtils = new ReplicationUtility(awsS3Client);
+
+const srcBucket = `source-bucket-${Date.now()}`;
+const awsDestBucket = process.env.AWS_S3_BACKBEAT_BUCKET_NAME;
+const destAWSLocation = process.env.AWS_S3_BACKEND_DESTINATION_LOCATION;
+const destFailBucket = process.env.AWS_S3_FAIL_BACKBEAT_BUCKET_NAME;
+const destFailLocation = process.env.AWS_S3_FAIL_BACKEND_DESTINATION_LOCATION;
+const hex = crypto.createHash('md5')
+    .update(Math.random().toString())
+    .digest('hex');
+const keyPrefix = `${srcBucket}/${hex}`;
+const key = `${keyPrefix}/object-to-replicate-${Date.now()}`;
+const destKeyPrefix = `${srcBucket}/${keyPrefix}`;
+
+const REPLICATION_TIMEOUT = 300000;
+
+function getAndCheckResponse(path, expectedBody, cb) {
+    let shouldContinue = false;
+    return doWhilst(next =>
+        makeGETRequest(path, (err, res) => {
+            assert.ifError(err);
+            assert.strictEqual(res.statusCode, 200);
+            getResponseBody(res, (err, body) => {
+                assert.ifError(err);
+                shouldContinue =
+                    JSON.stringify(body) !== JSON.stringify(expectedBody);
+                if (shouldContinue) {
+                    return setTimeout(next, 100);
+                }
+                return next();
+            });
+        }),
+    () => shouldContinue, cb);
+}
+
+function getPendingPath(location) {
+    const pathPrefix = '/_/backbeat/api/metrics/crr';
+    return `${pathPrefix}/${location}/pending`;
+}
+
+describe('Backbeat API pending metrics', function() {
+    this.timeout(REPLICATION_TIMEOUT);
+    const roleArn = 'arn:aws:iam::root:role/s3-replication-role';
+    const expectedBody = {
```

Comment:
Probably not a big issue since you set `expectedBody.pending.results` for every test, but it would be better practice to use `Object.assign` to copy `expectedBody` in each of the tests, so they aren't all accessing a global variable that could get mutated